### DISCUSSION TITLE
Nested package dependencies

### DISCUSSION THREAD

**Discussion by mhuzzell** [IBAHCM/RPiR#554] [Q&A for project] [answered]
If we use a package in Diversity that is already a dependency of BCI (and thus imported via BCI), should we also list it separately as a dependency of Diversity?

**Comment by richardreeve** [comment, marked as answer]
@mhuzzell - sorry for the delay in responding. You only need to add it as a dependency of Diversity if you use it directly in the Diversity package itself. If it is only used by BCI, then you can ignore it in Diversity.

**Reply by mhuzzell** [reply]
Thanks, and sorry for my own delay in clarifying the question. I mean if, e.g., we use the package 'vegan' directly in the Diversity package. 'vegan' is already a dependency of the BCI package, so when BCI is loaded, it loads vegan along with it -- and BCI is already a dependency of Diversity. Does 'vegan' also need to be listed as a dependency, in that case?

**Reply by richardreeve** [reply]
As I said earlier you "only need to add it as a dependency of Diversity if you use it directly in the Diversity package itself". If you use it in the Diversity package, it should be a direct dependency of Diversity. If not, not.
